lib/ohai/plugins/joyent.rb

```diff
@@ -25,24 +25,11 @@
   depends 'os', 'platform', 'virtualization'
 
   def collect_product_file
-    lines = []
-    if ::File.exists?("/etc/product")
-      ::File.open("/etc/product") do |file|
-        while line = file.gets
-          lines << line
-        end
-      end
-    end
-    lines
+    ::File.read("/etc/product").strip.split("\n") rescue []
   end
 
   def collect_pkgsrc
-    if File.exist?('/opt/local/etc/pkg_install.conf')
-      sm_pkgsrc = ::File.read("/opt/local/etc/pkg_install.conf").split("=")
-      sm_pkgsrc[1].chomp
-    else
-      nil
-    end
+    ::File.read("/opt/local/etc/pkg_install.conf").strip.split("=", 2).last rescue nil
```

Comment:
ah ok, i wasn't aware of `ohai` capturing all exceptions.

would it be best to remove the `rescue` statement completely then?  I think it's just as much of a problem if `/opt/local/etc/pkg_install.conf` doesn't exist, as it is if it contains garbage data.